packages/scaffold/src/modal/w3m-modal/index.ts

```diff
@@ -167,19 +168,38 @@ export class W3mModal extends LitElement {
     if (!this.connected || this.loading) {
       return
     }
-    const hasNetworkChanged = this.caipAddress && this.caipAddress !== caipAddress
 
+    const previousAddress = CoreHelperUtil.getPlainAddress(this.caipAddress)
+    const newAddress = CoreHelperUtil.getPlainAddress(caipAddress)
+    const previousNetworkId = CoreHelperUtil.getNetworkId(this.caipAddress)
+    const newNetworkId = CoreHelperUtil.getNetworkId(caipAddress)
     this.caipAddress = caipAddress
+
     if (this.isSiweEnabled) {
       const { SIWEController } = await import('@web3modal/siwe')
       const session = await SIWEController.getSession()
 
-      if (session && hasNetworkChanged) {
-        await SIWEController.signOut()
-        this.onSiweNavigation()
-      } else {
-        this.onSiweNavigation()
+      // If the address has changed and signOnAccountChange is enabled, sign out
+      if (session && previousAddress && newAddress && previousAddress !== newAddress) {
+        if (SIWEController.state._client?.options.signOutOnAccountChange) {
+          await SIWEController.signOut()
+          this.onSiweNavigation()
+        }
+
+        return
+      }
+
+      // If the network has changed and signOnNetworkChange is enabled, sign out
+      if (session && previousNetworkId && newNetworkId && previousNetworkId !== newNetworkId) {
+        if (SIWEController.state._client?.options.signOutOnNetworkChange) {
+          await SIWEController.signOut()
+          this.onSiweNavigation()
+        }
+
+        return
       }
+
+      this.onSiweNavigation()
```

Comment:
If not one of them, it's initialized from undefined to valid address so redirect to SIWE